### ISSUE TITLE
jspm install fails when tag contains a slash

### ISSUE THREAD

**Issue by mbreedlove** [jspm/jspm-cli#1871]
Here's the command that fails:

```
$ jspm install jquery-flexslider=github:woothemes/FlexSlider@version/2.1

    Looking up github:woothemes/FlexSlider

    err  Unable to resolve version version2F2.1% for woothemes/FlexSlider.

    warn Installation changes not saved.
```

[`Package.getVersionMatch`](https://github.com/jspm/jspm-cli/blob/master/lib/package.js#L138) seems to work correctly, returning `new PackageName(pkg.name + '@' + lookupObj.version, true)` 

However, because the second parameter is true, it escapes the version, resulting in 

```
{ exactName: 'github:woothemes/FlexSlider@version%2F2.1',
  registry: 'github',
  package: 'woothemes/FlexSlider',
  name: 'github:woothemes/FlexSlider',
  version: 'version%2F2.1',
  exactPackage: 'woothemes/FlexSlider@version%2F2.1' }
```

 This is a problem during [`Package.download`](https://github.com/jspm/jspm-cli/blob/master/lib/package.js#L465) when we try and get the `lookupObj`: 

`var lookupObj = lookups[pkg.package][pkg.version];`

None of the keys in `lookups[pkg.package]` are escaped: 

```
{ master:
   { hash: '622f7fda936f87ce8c95a437bba07162de35e0d3',
     meta: {},
     stable: false,
     version: 'master' },
...
  'version/2.1':
   { hash: '54e6d318cc93d5411e293b16045b929b91277187',
     meta: {},
     version: 'version/2.1' },
...
```

So the lookup fails. 

We should be able to safely `decodeURIComponent()` on `pkg.version` in all cases. 

`var lookupObj = lookups[pkg.package][decodeURIComponent(pkg.version)];`

However, maybe there is a problem with this, I don't really understand the motivation for escaping the tags anyway.

If that is an acceptable solution, I'll happily make a pull request.

**Comment by guybedford** [comment]
Is this on jspm 0.16 or the 0.17 beta? I believe this is a bug in 0.16 and fixed in the beta.

**Comment by mbreedlove** [comment]
This happened on 0.16.35. 

I just upgraded to 0.17.0-beta.17, ran `jspm install jquery-flexslider=github:woothemes/FlexSlider@version/2.1`, the package resolved to the master branch rather than the "version/2.1" tag.

I tried modifying jspm.js and package.json like I mentioned in my first posting. 

```
SystemJS.config({
  map: {
    "jquery-flexslider": "github:woothemes/FlexSlider@version/2.1"
  }
});
```

```
{
  "devDependencies": {
    "jspm": "^0.17.0-beta"
  }
  "jspm": {
    "peerDependencies": {
      "jquery-flexslider": "github:woothemes/FlexSlider@version/2.1"
    }
  }
}
```

```
$ jspm install
(node:65708) fs: re-evaluating native module sources is not supported. If you are using the graceful-fs module, please update it to a more recent version.
     Downloading github:woothemes/FlexSlider@version/2.1
warn Package github:woothemes/FlexSlider@version/2.1 has no "main" entry point set in its package config.
ok   Installed peer jquery-flexslider to github:woothemes/FlexSlider@version%2F2.1 (version/2.1)
     Removing package files for github:woothemes/FlexSlider@version
     Install tree has no forks.

ok   Install complete.
```

Looks like there's still an escaping issue...

Here's my jspm.js after the `jspm install` with modified jspm.js and package.json.

```
{
  "devDependencies": {
    "jspm": "^0.17.0-beta"
  }
  "jspm": {
    "peerDependencies": {
      "jquery-flexslider": "github:woothemes/FlexSlider@version%2F2.1"
    }
  }
}
```

Notice the escaped tag. The package was also not present in jspm_packages/github.

If you have any other ideas, I'll gladly try and post the output here.

**Comment by guybedford** [comment]
Yes the `/` will be turned into a `%2F` in the file system to support it, but the folder `FlexSlider@version%2F2.1` should exist in 0.17.

**Comment by guybedford** [comment]
Actually, yeah this escaping is out again.

**Comment by guybedford** [comment]
Fixed in https://github.com/jspm/jspm-cli/commit/3af0e65c1a5defb468aff5ddc22cd3c3fddcdf50.

**Comment by guybedford** [comment]
Released in 0.17.0-beta.18.
